packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -466,17 +466,71 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
         // snapshot code path. We should leverage the fact that these deltas are returned to speed up the deltas fetch.
         const { headers, url } = getUrlAndHeadersWithAuth(`${this.snapshotUrl}/trees/latest${snapshotOptions}`, storageToken);
 
+        let isOptionsCall = false;
+
+        if (Object.keys(headers).length) {
+            isOptionsCall = true;
+        }
+
         // This event measures only successful cases of getLatest call (no tokens, no retries).
         const { snapshot, canCache } = await PerformanceEvent.timedExecAsync(this.logger, { eventName: "TreesLatest" }, async (event) => {
+            const startTime = performance.now();
```

Comment:
This is not isomorphic, since this is not available in node environment. We should instead use the isomorphic performance from our @fluidframework/common-utils package. @heliocliu for more info, or correct me if I'm mistaken.

Reply:
Right, we should use this instead:
`import { performanceNow } from "@fluidframework/common-utils";`
Though it looks like additional performance APIs are used below which are not covered in our isomorphic wrapper.  This code will fail running in node, and if this code is only ever expected to run in browser we need to make sure that's clear somehow.  If `getEntriesByType` functionality is absolutely needed here and this also needs to run in both environments, we need to update the platform abstraction in common-utils

Reply:
@heliocliu Thank you for the inputs. The getEntreisByType() is needed as we need to find the right resource, if there are alternatives available, we can explore that. Also, I'm not clear on the code running in node and browser only, do you mean the current code will cause issues and and we need to add a check here? Any pointers to similar check done elsewhere? @ram-ramasubra @jatgarg FYI.

Reply:
I put in an issue #3654 to track this. I think node already has perf_hooks for a similarly behaving `performance` object.